provider/k8s.go

```diff
@@ -474,80 +477,64 @@ func (store genericBlobStore) Read(key string) ([]byte, error) {
 }
 
 func (store genericBlobStore) Serve(key string) (Iterator, error) {
-	r, err := store.bucket.NewReader(ctx, key, nil)
+	b, err := store.bucket.ReadAll(ctx, key)
```

Comment:
For 1M row files I assume this is a bottleneck, as it would have been in the original version too.  Should the server in the future read chunks of bytes at a time so it doesn't need to read everything before writing?

Reply:
The other implementation did this too. So far theres no issues with it